docs/cookbook/recipe_sonata_admin_without_user_bundle.rst

```diff
@@ -0,0 +1,451 @@
+SonataAdmin without FOSUserBundle/SonataUserBundle
+==================================================
+
+Here we will explain how you can use SonataAdmin with Symfony Guard instead
+of using FOSUserBundle or SonataUserBundle.
+
+Code used in this short guide can be found `here <https://github.com/kunicmarko20/guard-talk-example>`__
+with support for different Symfony versions.
+
+.. note::
+    This article assumes you are using Symfony 4.
+
+The recipe
+----------
+
+Things we will need to do:
+
+- User Entity
+- UserProvider
+- LoginForm
+- LoginAuthenticator
+- LoginController
+- Setup firewall in `security.yaml`
+- Login template resembling Sonata style
+
+User Entity
+^^^^^^^^^^^
+
+This represents your security user, you can read more about it `here <http://symfony.com/doc/current/security/entity_provider.html#create-your-user-entity>`__.
+
+.. code-block:: php
+
+    namespace App\Entity;
+
+    use Doctrine\ORM\Mapping as ORM;
+    use Symfony\Component\Security\Core\User\UserInterface;
+
+    /**
+     * @ORM\Entity
+     * @ORM\Table(name="user")
+     */
+    class User implements UserInterface
+    {
+        /**
+         * @ORM\Id
+         * @ORM\Column(type="integer")
+         * @ORM\GeneratedValue(strategy="AUTO")
+         */
+        private $id;
+
+        /**
+         * @ORM\Column(name="email", type="string", unique=true)
+         */
+        private $email;
+
+        /**
+         * @ORM\Column(name="password", type="string", nullable=true)
+         */
+        private $password;
+
+        /**
+         * @ORM\Column(name="roles", type="json_array")
+         */
+        private $roles;
+
+        //...
+    }
+
+UserProvider
+^^^^^^^^^^^^
+
+This represents your user provider, it will be used to load your security users, read
+more about it `here <http://symfony.com/doc/current/security/custom_provider.html#create-a-user-provider>`__.
+
+.. code-block:: php
+
+    namespace App\Security;
+
+    use App\Entity\User;
+    use Doctrine\ORM\EntityManagerInterface;
+    use Symfony\Component\Security\Core\User\UserInterface;
+    use Symfony\Component\Security\Core\User\UserProviderInterface;
+    use Symfony\Component\Security\Core\Exception\UsernameNotFoundException;
+
+    class UserProvider implements UserProviderInterface
+    {
+        /**
+         * @var EntityManagerInterface
+         */
+        private $entityManager;
+
+        public function __construct(EntityManagerInterface $entityManager)
```

Comment:
```
- $entityManager
+ $em
```

or `$om`?

Reply:
ok, I read that Symfony itself will move from the `$entityManager` wording, but not sure.

//cc @greg0ire ??

Reply:
Since I did a refactoring of the docs a few months ago https://github.com/symfony/symfony-docs/pull/9354, I hope they don't go back to `$em` 😄

Reply:
ah good to know :)

then everything is fine 👍

Reply:
phpmd whines when using small variable names, and abbreviations have a bad reputation. I'd say avoid them stick with this code.